### ISSUE TITLE
`make RELEASE=1` fails after vTPM merge

### ISSUE THREAD

**Issue by joergroedel** [coconut-svsm/svsm#313]
I run into build errors with release builds using the latest main branch. The error message is:

```
error: linking with `ld` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/joro/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/joro/.cargo/bin:/home/joro/bin:/usr/local/bin:/usr/bin:/bin" VSLANG="1033" "ld" "/tmp/rustcquPRDt/symbols.o" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/stage2-d4977d5895cc0683.stage2.b6c9f965d90ae170-cgu.0.rcgu.o" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/stage2-d4977d5895cc0683.1oneoujy45tomked.rcgu.o" "--as-needed" "-L" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps" "-L" "/home/joro/src/svsm/target/release/deps" "-L" "/home/joro/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-none/lib" "-Bstatic" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libsvsm-2f1d75f89eae45ca.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libelf-9dbc3f5a91e4001b.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/liblog-199db79b6ff6d0ad.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/liblibmstpm-e4ea92b778d7d044.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libintrusive_collections-ac544836f742b2a0.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libmemoffset-4670a9c19b1fe444.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libigvm_defs-01bef8bf3ca49a1f.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libstatic_assertions-3eadaa91c045972a.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libopen_enum-b63b4af2d5bee7bf.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libpackit-dd1f12a3a62e061b.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libzerocopy-f5c7ca2b226190a6.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libaes_gcm-02ef613850b3f059.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libctr-8a42ff1ee0f8d6e5.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libghash-526b75ac36c61266.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libpolyval-56a955881f1290f5.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libopaque_debug-6ab9204f271d0a99.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libuniversal_hash-e99b27a4a6311f39.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libsubtle-c41c5dbb07f37b99.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libaes-ed144db2af766958.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libcfg_if-fc125a0028f81851.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libcipher-a7fbdf895f18b8a4.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libinout-832be430908771d2.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libaead-0f20ce055fa711ea.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libcrypto_common-68a3e50ed252b368.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libgeneric_array-f95f4cc78c8ed14f.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libtypenum-c8172db6c662b867.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libbitflags-a5808a034eae0866.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libcpuarch-b61687b4b56fc29c.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libbootlib-507b6b8d6a0dfbcd.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libzerocopy-c7ea11b0970ffbd5.rlib" "/home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libbyteorder-1695cc72465b0ba8.rlib" "/home/joro/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-none/lib/liballoc-03d92df9b943c3f1.rlib" "/home/joro/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-none/lib/librustc_std_workspace_
core-4c552adff567c780.rlib" "/home/joro/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-none/lib/libcore-841fb3f9d3200020.rlib" 
"/home/joro/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-none/lib/libcompiler_builtins-6ef5172fdae4aa93.rlib" "-Bdynamic" "--
eh-frame-hdr" "-z" "noexecstack" "-L" "/home/joro/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-none/lib" "-o" "/home/joro/src
/svsm/target/x86_64-unknown-none/release/deps/stage2-d4977d5895cc0683" "--gc-sections" "-pie" "-z" "relro" "-z" "now" "-O1" "-nostdlib" "--build-id=none" "-Tk
ernel/src/stage2.lds" "-no-pie"
  = note: ld: /home/joro/src/svsm/target/x86_64-unknown-none/release/deps/libsvsm-2f1d75f89eae45ca.rlib(svsm-2f1d75f89eae45ca.svsm.5dac7fa7ac709030-cgu.08.rcg
u.o): in function `<svsm::vtpm::mstpm::MsTpm as svsm::vtpm::MsTpmSimulatorInterface>::send_tpm_command':
          svsm.5dac7fa7ac709030-cgu.08:(.text._ZN80_$LT$svsm..vtpm..mstpm..MsTpm$u20$as$u20$svsm..vtpm..MsTpmSimulatorInterface$GT$16send_tpm_command17h4a0698
dba961caa4E+0xe2): undefined reference to `_plat__LocalitySet'
          ld: svsm.5dac7fa7ac709030-cgu.08:(.text._ZN80_$LT$svsm..vtpm..mstpm..MsTpm$u20$as$u20$svsm..vtpm..MsTpmSimulatorInterface$GT$16send_tpm_command17h4a
0698dba961caa4E+0xfa): undefined reference to `_plat__RunCommand'
          
  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scr
ipts.html#rustc-link-lib)

error: could not compile `svsm` (bin "stage2") due to 1 previous error
```

I don't see the same issue with debug builds.

**Comment by roy-hopkins** [comment]
I think the simplest solution is to just prevent building the mstpm feature (and in fact, all other features) for stage 2. I've created a PR to do this: https://github.com/coconut-svsm/svsm/pull/316

**Comment by cclaudio** [comment]
Issue fixed by PRs #316 and #317
